packages/react/src/components/CreateAnimation.tsx

```diff
@@ -0,0 +1,159 @@
+import { Animation, AnimationCallbackOptions, AnimationDirection, AnimationFill, AnimationKeyFrames, AnimationLifecycle, createAnimation } from '@ionic/core';
+import React from 'react';
+
+interface PartialPropertyValue { property: string; value: any; }
+interface PropertyValue { property: string; fromValue: any; toValue: any; }
+
+export interface CreateAnimationProps {
+  delay?: number;
+  direction?: AnimationDirection;
+  duration?: number;
+  easing?: string;
+  fill?: AnimationFill;
+  iterations?: number;
+  id?: string;
+
+  afterAddRead?: () => void;
+  afterAddWrite?: () => void;
+  afterClearStyles?: string[];
+  afterStyles?: { [property: string]: any };
+  afterAddClass?: string | string[];
+  afterRemoveClass?: string | string[];
+
+  beforeAddRead?: () => void;
+  beforeAddWrite?: () => void;
+  beforeClearStyles?: string[];
+  beforeStyles?: { [property: string]: any };
+  beforeAddClass?: string | string[];
+  beforeRemoveClass?: string | string[];
+
+  onFinish?: { callback: AnimationLifecycle; opts?: AnimationCallbackOptions; };
+
+  keyframes?: AnimationKeyFrames;
+  from?: PartialPropertyValue;
+  to?: PartialPropertyValue;
+  fromTo?: PropertyValue;
+
+  play?: boolean;
+  pause?: boolean;
+  stop?: boolean;
+  destroy?: boolean;
+
+  progressStart?: { forceLinearEasing: boolean, step?: number };
+  progressStep?: { step: number };
+  progressEnd?: { playTo: 0 | 1 | undefined, step: number, dur?: number };
+}
+
+export class CreateAnimation extends React.Component<CreateAnimationProps> {
+  private nodes: Map<number, HTMLElement> = new Map();
+  animation?: Animation;
+
+  updateAnimation(props: any) {
+    const animation = this.animation;
+    if (animation === undefined) { return; }
+
+    if (this.nodes.size > 0) {
```

Comment:
this method of adding the child node to the animation probably won't work if someone ever adds children to the animation dynamically. Might be a future change as thats probably an edge case, but sooner or later someone will file a bug :)